engine/crates/runtime-local/src/fetch.rs

```diff
@@ -10,28 +10,49 @@ use self::websockets::StreamingRequest;
 
 pub struct NativeFetcher {
     client: reqwest::Client,
+    /// Timeout configuration per subgraph
+    timeouts: HashMap<String, Duration>,
```

Comment:
I am wondering: does this belong here, or one level down in the engine v2 config?

Reply:
the timeout config should be in `engine_v2::Config` IMO and intuitively would pass it to the fetcher when requesting data. With defaultss being baked into the fetcher at creation.

This would be also easier to test with integration-tests and ensure this also works with cloudflare and other platforms.

Reply:
done